lib/util.js

```diff
@@ -803,7 +813,7 @@ function formatProperty(ctx, value, recurseTimes, visibleKeys, key, array) {
       str = ctx.stylize('[Setter]', 'special');
     }
   }
-  if (!hasOwnProperty(visibleKeys, key)) {
+  if (visibleKeys[key] === undefined) {
```

Comment:
It may be worth to eventually make `visibleKeys` an ES2015 `Set` instead of emulating it via an object, but I don't know how it will affect performance with the current V8 (if it will at all).